cmd/lakefs/cmd/kv.go

```
@@ -156,6 +161,175 @@ var kvScanCmd = &cobra.Command{
 	},
 }
 
+var kvDumpCmd = &cobra.Command{
+	Use:   "dump",
+	Short: "Dump KV store data to JSON format",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cfg := LoadConfig().GetBaseConfig()
+
+		outputFile, err := cmd.Flags().GetString("output")
+		if err != nil {
+			return err
+		}
+
+		sectionsFlag, err := cmd.Flags().GetString("sections")
+		if err != nil {
+			return err
+		}
+
+		pretty, err := cmd.Flags().GetBool("pretty")
+		if err != nil {
+			return err
+		}
+
+		// Parse sections (comma-separated)
+		var sections []string
+		if sectionsFlag != "" {
+			sections = strings.Split(sectionsFlag, ",")
+			for i, s := range sections {
+				sections[i] = strings.TrimSpace(s)
+			}
+		}
+		// Empty default means all supported sections
+
+		ctx := cmd.Context()
+		kvParams, err := kvparams.NewConfig(&cfg.Database)
+		if err != nil {
+			return fmt.Errorf("KV params: %w", err)
+		}
+
+		kvStore, err := kv.Open(ctx, kvParams)
+		if err != nil {
+			return fmt.Errorf("failed to open KV store: %w", err)
+		}
+		defer kvStore.Close()
+
+		// Create dump
+		dump, err := kv.CreateDump(ctx, kvStore, sections)
+		if err != nil {
+			return fmt.Errorf("failed to create dump: %w", err)
+		}
+
+		// Encode to JSON
+		var output *os.File
+		if outputFile == "" || outputFile == "-" {
+			output = os.Stdout
+		} else {
+			output, err = os.Create(outputFile)
+			if err != nil {
+				return fmt.Errorf("failed to create output file: %w", err)
+			}
```

Consider failing fast (before dump)